lib/demos/material/menu_demo.dart

```diff
@@ -277,37 +280,80 @@ class _SimpleMenuDemoState extends State<_SimpleMenuDemo> {
 // Pressing the PopupMenuButton on the right of this item shows a menu
 // whose items have checked icons that reflect this app's state.
 class _ChecklistMenuDemo extends StatefulWidget {
-  const _ChecklistMenuDemo({Key key, this.showInSnackBar}) : super(key: key);
+  const _ChecklistMenuDemo({
+    Key key,
+    @required this.restorationId,
+    this.showInSnackBar,
+  }) : super(key: key);
 
+  final String restorationId;
   final void Function(String value) showInSnackBar;
 
   @override
   _ChecklistMenuDemoState createState() => _ChecklistMenuDemoState();
 }
 
-class _ChecklistMenuDemoState extends State<_ChecklistMenuDemo> {
-  List<CheckedValue> _checkedValues;
+class _RestorableCheckedValue extends RestorableValue<Set<int>> {
```

Comment:
Wondering what you think about this @goderbauer and if you have any suggestions for improving this.

Reply:
My suggestion would be to extend RestorableProperty directly and write a class that looks something like this:

```dart
class _RestorableCheckedValues extends RestorableProperty<List<int>> {
 Set<CheckedValue> _checked = Set<>();

  void check(CheckedValue value) {
    _checked.add(value);
   notifyListeners();
  }

  void uncheck(CheckedValue value) {
    _checked.remove(value);
   notifyListeners();
  }
  
  void isChecked(CheckedValue value) {
    return _checked.contains(value);
  }

  void initWithValue(Set<CheckedValue> a) {
    _checked = a;
  }

  @override
  Object toPrimitives() {
    return _checked.map((a) => e.index).toList();
  }

  @override
  Set<int> fromPrimitives(Object data) {
    final checkedValues = data as List<dynamic>;
    return Set.from(checkedValues.map<int>((dynamic id) => CheckedValues.values[id as int]));
  }

}
```

(this code probably doesn't compile, but you get the idea)

That way, you get a nice API to use within _ChecklistMenuDemoState.

Looking at the code below, instead of check/uncheck you could also have a simple toggle method. And you probably need to exposes a map function which it takes a callback and iterates over the values in _checked.